<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 example - Human gut usage | ICME 13 - Rome Edition</title>
  <meta name="description" content="Book for for the ICME series courses" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="5 example - Human gut usage | ICME 13 - Rome Edition" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Book for for the ICME series courses" />
  <meta name="github-repo" content="microbeco/microbeco.github.io" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 example - Human gut usage | ICME 13 - Rome Edition" />
  
  <meta name="twitter:description" content="Book for for the ICME series courses" />
  

<meta name="author" content="Manuela Coci" />
<meta name="author" content="Luigi Gallucci" />
<meta name="author" content="Davide Corso" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="metagenomics.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<div style="text-align:center">
<li class="toc-logo"><a href="./"><img src="img/icme_13.png" width="250" height="250"></a></li>
<li><a href="https://www.microbeco.org/icme13-roma-2024/"><strong>ICME 13</strong></a></li>
</div>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#icme-13---rome"><i class="fa fa-check"></i><b>1.1</b> ICME 13 - Rome</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>2</b> Introduction to the command line</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#set-up-a-terminal"><i class="fa fa-check"></i><b>2.1</b> Set-up a terminal</a></li>
<li class="chapter" data-level="2.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#working-with-the-command-line"><i class="fa fa-check"></i><b>2.2</b> Working with the command line</a></li>
<li class="chapter" data-level="2.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#playing-around-with-basic-unix-commands"><i class="fa fa-check"></i><b>2.3</b> Playing around with basic UNIX commands</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#some-notes"><i class="fa fa-check"></i><b>2.3.1</b> Some notes!</a></li>
<li class="chapter" data-level="2.3.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-and-navigating-directories"><i class="fa fa-check"></i><b>2.3.2</b> Creating and navigating directories</a></li>
<li class="chapter" data-level="2.3.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-a-new-file"><i class="fa fa-check"></i><b>2.3.3</b> Creating a new file</a></li>
<li class="chapter" data-level="2.3.4" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#copying-renaming-moving-and-deleting-files"><i class="fa fa-check"></i><b>2.3.4</b> Copying, renaming, moving and deleting files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="s-analysis.html"><a href="s-analysis.html"><i class="fa fa-check"></i><b>3</b> 16s Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="s-analysis.html"><a href="s-analysis.html#commands-and-features-that-we-will-use-in-our-practice"><i class="fa fa-check"></i><b>3.1</b> Commands and features that we will use in our practice</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="s-analysis.html"><a href="s-analysis.html#touch"><i class="fa fa-check"></i><b>3.1.1</b> <code>touch</code></a></li>
<li class="chapter" data-level="3.1.2" data-path="s-analysis.html"><a href="s-analysis.html#echo"><i class="fa fa-check"></i><b>3.1.2</b> <code>echo</code></a></li>
<li class="chapter" data-level="3.1.3" data-path="s-analysis.html"><a href="s-analysis.html#cat"><i class="fa fa-check"></i><b>3.1.3</b> <code>cat</code></a></li>
<li class="chapter" data-level="3.1.4" data-path="s-analysis.html"><a href="s-analysis.html#section"><i class="fa fa-check"></i><b>3.1.4</b> <code>$</code></a></li>
<li class="chapter" data-level="3.1.5" data-path="s-analysis.html"><a href="s-analysis.html#chmod"><i class="fa fa-check"></i><b>3.1.5</b> <code>chmod</code></a></li>
<li class="chapter" data-level="3.1.6" data-path="s-analysis.html"><a href="s-analysis.html#basename"><i class="fa fa-check"></i><b>3.1.6</b> <code>basename</code></a></li>
<li class="chapter" data-level="3.1.7" data-path="s-analysis.html"><a href="s-analysis.html#create-a-shell-scripts"><i class="fa fa-check"></i><b>3.1.7</b> Create a shell scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="s-analysis.html"><a href="s-analysis.html#package-manager"><i class="fa fa-check"></i><b>3.2</b> Package Manager</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="s-analysis.html"><a href="s-analysis.html#conda"><i class="fa fa-check"></i><b>3.2.1</b> Conda</a></li>
<li class="chapter" data-level="3.2.2" data-path="s-analysis.html"><a href="s-analysis.html#mamba-recommended"><i class="fa fa-check"></i><b>3.2.2</b> Mamba (Recommended)</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="s-analysis.html"><a href="s-analysis.html#base-calling"><i class="fa fa-check"></i><b>3.3</b> Base Calling</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="s-analysis.html"><a href="s-analysis.html#guppy"><i class="fa fa-check"></i><b>3.3.1</b> Guppy</a></li>
<li class="chapter" data-level="3.3.2" data-path="s-analysis.html"><a href="s-analysis.html#dorado"><i class="fa fa-check"></i><b>3.3.2</b> Dorado</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="s-analysis.html"><a href="s-analysis.html#filtering-and-trimming"><i class="fa fa-check"></i><b>3.4</b> Filtering and Trimming</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="s-analysis.html"><a href="s-analysis.html#nanofilt"><i class="fa fa-check"></i><b>3.4.1</b> NanoFilt</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="s-analysis.html"><a href="s-analysis.html#subsetting"><i class="fa fa-check"></i><b>3.5</b> Subsetting</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="s-analysis.html"><a href="s-analysis.html#bbmap-tools"><i class="fa fa-check"></i><b>3.5.1</b> BBMAP Tools</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="s-analysis.html"><a href="s-analysis.html#taxonomic-assignment"><i class="fa fa-check"></i><b>3.6</b> Taxonomic Assignment</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="s-analysis.html"><a href="s-analysis.html#emu"><i class="fa fa-check"></i><b>3.6.1</b> EMU</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="metagenomics.html"><a href="metagenomics.html"><i class="fa fa-check"></i><b>4</b> Metagenomics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="metagenomics.html"><a href="metagenomics.html#setup-environments-for-this-pipeline"><i class="fa fa-check"></i><b>4.1</b> Setup Environments for this pipeline</a></li>
<li class="chapter" data-level="4.2" data-path="metagenomics.html"><a href="metagenomics.html#reads-quality-check"><i class="fa fa-check"></i><b>4.2</b> Reads Quality Check</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="metagenomics.html"><a href="metagenomics.html#bbmap-stats.sh"><i class="fa fa-check"></i><b>4.2.1</b> BBMap / <code>stats.sh</code></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="metagenomics.html"><a href="metagenomics.html#assembly"><i class="fa fa-check"></i><b>4.3</b> Assembly</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="metagenomics.html"><a href="metagenomics.html#flye-metaflye"><i class="fa fa-check"></i><b>4.3.1</b> Flye (Metaflye)</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="metagenomics.html"><a href="metagenomics.html#read-mapping"><i class="fa fa-check"></i><b>4.4</b> Read mapping</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="metagenomics.html"><a href="metagenomics.html#minimap2"><i class="fa fa-check"></i><b>4.4.1</b> Minimap2</a></li>
<li class="chapter" data-level="4.4.2" data-path="metagenomics.html"><a href="metagenomics.html#samtools"><i class="fa fa-check"></i><b>4.4.2</b> samtools</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="metagenomics.html"><a href="metagenomics.html#binning"><i class="fa fa-check"></i><b>4.5</b> Binning</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="metagenomics.html"><a href="metagenomics.html#semibin2"><i class="fa fa-check"></i><b>4.5.1</b> SemiBin2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="example---human-gut-usage.html"><a href="example---human-gut-usage.html"><i class="fa fa-check"></i><b>5</b> example - Human gut usage</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ICME 13 - Rome Edition</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example---human-gut-usage" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> example - Human gut usage<a href="example---human-gut-usage.html#example---human-gut-usage" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>–sequencing-type long_read
–environment human_gut
–input-fasta AD/assembly.fasta
–input-bam aln.bam
–output output</p>
<div id="section-2" class="section level27" number="5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1">
<p class="heading"><span class="header-section-number">5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
<p>Mags quality check</p>
<p>As well as the original reads, also High Quality MAGs were checked</p>
<p>CHECKM2</p>
<p>CHECKM2 - Rapid assessment of genome bin quality using machine learning
<a href="https://github.com/chklovski/CheckM2" class="uri">https://github.com/chklovski/CheckM2</a></p>
<p>CheckM2 has universally trained machine learning models it applies regardless of taxonomic lineage to predict the completeness and contamination of genomic bins.
Completeness is the percentage of the mapped genome that were covered by each mag.
Contamination is the inclusion of foreign sequences on the mags</p>
<p>The strain heterogeneity (SH) index indicates the proportion of the contamination that appears to be from the same or similar strains (as determined with an AAI threshold).</p>
<p>In order to extract the completeness and contamination for each Mags, You will also need to download and install the external DIAMOND database that CheckM2 relies on for rapid annotation.</p>
<p>checkm2 database</p>
<p>Options used
–download
–path /path/to/DB</p>
<p>The database path can also be set by setting the environmental variable CHECKM2DB using:
export CHECKM2DB=“path/to/database”</p>
<p>The main use of CheckM2 is to predict the completeness and contamination of metagenome-assembled genomes (MAGs)</p>
<p>checkm2 predict</p>
<p>Options used
–input /path/output/output_bins/
–output-directory output</p>
</div>
<div id="section-3" class="section level22" number="5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1">
<p class="heading"><span class="header-section-number">5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
<p>Taxonomic assingment</p>
<p>GTDB-Tk
GTDB-tk - assigning taxonomic classifications
<a href="https://github.com/Ecogenomics/GTDBTk" class="uri">https://github.com/Ecogenomics/GTDBTk</a></p>
<p>GTDB-Tk is the software toolkit used for assigning objective taxonomic classifications to bacterial and archaeal genomes based on the Genome Database Taxonomy (GTDB).
It is designed to work with recent advances that allow hundreds or thousands of metagenome-assembled genomes (MAGs) to be obtained directly from environmental samples.</p>
<p>GTDB-Tk requires an external database that needs to be downloaded and unarchived:
DB <a href="https://ecogenomics.github.io/GTDBTk/installing/index.html#gtdb-tk-reference-data" class="uri">https://ecogenomics.github.io/GTDBTk/installing/index.html#gtdb-tk-reference-data</a></p>
<p>The classify workflow (classify_wf) with gtdbk consists of four steps: ani_screen, identify, align, and classify</p>
<p>The identify step calls genes using Prodigal, and uses HMM models and the HMMER package to identify the 120 bacterial and 53 archaeal marker genes used for phylogenetic inference (Parks et al., 2018). Multiple sequence alignments (MSA) are obtained by aligning marker genes to their respective HMM model.</p>
<p>GTDB-Tk classifies each genome based on its placement in the reference tree, its relative evolutionary divergence, and/or average nucleotide identity (ANI) to reference genomes.</p>
<p>The align step concatenates the aligned marker genes and filters the concatenated MSA to approximately 5,000 amino acids.</p>
<p>Finally, the classify step uses pplacer to find the maximum-likelihood placement of each genome in the GTDB-Tk reference tree. GTDB-Tk classifies each genome based on its placement in the reference tree, its relative evolutionary divergence, and/or average nucleotide identity (ANI) to reference genomes.</p>
<p>gtdbtk classify_wf
Options used
–genome_dir directory containing genome files in FASTA format<br />
–out_dir directory to output files
–extension fa extension of files to process, gz = gzipped
–force continue processing if an error occurs on a single genome
–skip_ani_screen Skip the ani_screening step to classify genomes using mash and FastANI
–pplacer_cpus 32 number of CPUs to use during pplacer placement</p>
<p>list of output files:
1
.summary.tsv: Classifications for bacterial and archaeal genomes (see the GTDB-Tk documentation for details.
.classify.tree.gz: Reference tree in Newick format containing query genomes placed with pplacer.
.markers_summary.tsv: A summary of unique, duplicated, and missing markers within the 120 bacterial marker set, or the 122 archaeal marker set for each submitted genome.
.msa.fasta.gz: FASTA file containing MSA of submitted and reference genomes.
.filtered.tsv: A list of genomes with an insufficient number of amino acids in MSA.
.log: Log files.
.failed_genomes.tsv: A list of genomes for which the GTDB-Tk analysis failed, e.g. because Prodigal could not detect any genes.</p>
<p>2
gtdbtk_summary.tsv: A summary table of the GTDB-Tk classification results for all bins</p>
<p>The taxonomic classification of each bacterial and archaeal genome is contained in the [prefix].[domain].summary.tsv output files.</p>
<p>in summary.tsv we find :
- fastani_reference: indicates the accession number of the reference genome (species) to which a user genome was assigned based on ANI and AF. ANI values are only calculated when a query genome is placed within a defined genus and are evaluated for all reference genomes in that genus.
- fastani_reference_radius: indicates the species-specific ANI circumscription radius of the reference genomes used to determine if a query genome should be classified to the same species as the reference.
<a href="https://ecogenomics.github.io/GTDBTk/files/summary.tsv.html" class="uri">https://ecogenomics.github.io/GTDBTk/files/summary.tsv.html</a>
- fastani_af: indicates the alignment fraction (AF) between the query and above reference genome.
- closest_placement_reference: indicates the accession number of the reference genome when a genome is placed on a terminal branch.
- classification_method: indicates the rule used to classify the genome.</p>
<p>A strain identifier is used as a placeholder for the genus name when there is no existing genus name and no binomially named representative genome.
This placeholder genus name is generally derived from the oldest representative genome within the lineage and formed from NCBI organism name or NCBI infraspecific/strain ID.</p>
<div id="section-4" class="section level23" number="5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.1">
<p class="heading"><span class="header-section-number">5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.1</span> </p>
<p>Anvi’o</p>
<p>For the final and detailed analysis of the Mags (both from short and long reads) several specific tool were developed in the last years (i.e. is Anvi’o).</p>
<p>Anvi’o is a comprehensive platform that brings together many aspects of today’s cutting-edge computational strategies of data-enabled microbiology, including genomics, metagenomics, metatranscriptomics, pangenomics, metapangenomics, phylogenomics, and microbial population genetics in an integrated and easy-to-use fashion through extensive interactive visualization capabilities.</p>
<p><a href="https://anvio.org/" class="uri">https://anvio.org/</a>
<a href="https://github.com/merenlab/anvio" class="uri">https://github.com/merenlab/anvio</a></p>
<p>As requirement, install the database PFAM, KEGG, SCG e CAZy.
anvi-setup-kegg-data –mode KOfam –only-download –kegg-data-dir /SERVER/anvio-db/kegg/
anvi-setup-pfams –pfam-data-dir /SERVER/anvio-db/pfam/
anvi-setup-cazymes –cazyme-data-dir /SERVER/anvio-db/cazy/
anvi-setup-scg-taxonomy –scgs-taxonomy-data-dir /SERVER/anvio-db/scg/ –reset</p>
<p>the following workflow is useful for converting a bunch of genomes into an anvi’o-compatible format. It generates contigs databases from each input FASTA file, and subsequently runs a variety of annotation programs of your choice to populate these databases with some useful information for your downstream work (i.e. functions, single-copy-core genes, taxonomy, etc).</p>
<p>To start things going with this workflow, first ask anvi’o to give you a default workflow-config file for the contigs workflow:</p>
<p>anvi-run-workflow
-w contigs select the different type of “workflow (i.e. contig, metagenome …)
–get-default-config contigs.json generate a default config file (.json) for a given workflow</p>
<p><a href="https://anvio.org/help/main/workflows/contigs/" class="uri">https://anvio.org/help/main/workflows/contigs/</a></p>
<p>Here create a file_list.tsv
echo -e “name” &gt; file_list.tsv &amp;&amp; find path/to/selected_genomes -type f -exec sh -c ‘echo -e “$(basename”{}” .fa)“’ ; &gt;&gt; file_list.tsv</p>
<p>If everything looks alright, you can run this workflow the following way from the same folder of file_list.tsv:</p>
<p>anvi-run-workflow
-w contigs
-c contigs.json
–additional-params
–jobs 3</p>
<p>Finally, perform the annotation using one of the pre-loaded database</p>
<p>for i in 02_CONTIGS/*.db; # 02_CONTIGS viene generata da anvi-run-workflow
do
echo “$i”;
anvi-run-cazymes -c $i –cazyme-data-dir path/to/CAZYdb/;
done</p>
<p>.. and export annotazione</p>
<p>for i in 02_CONTIGS/*.db; # 02_CONTIGS viene generata da anvi-run-workflow
do
echo “$i”;
anvi-export-functions -c $i
done</p>
<div id="section-5" class="section level26" number="5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.1.0.0.1">
<p class="heading"><span class="header-section-number">5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.1.0.0.1</span> </p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="metagenomics.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["icme13.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
