<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Our Metagenomic practice | ICME 13 - Rome Edition</title>
  <meta name="description" content="Book for for the ICME series courses" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Our Metagenomic practice | ICME 13 - Rome Edition" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Book for for the ICME series courses" />
  <meta name="github-repo" content="microbeco/microbeco.github.io" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Our Metagenomic practice | ICME 13 - Rome Edition" />
  
  <meta name="twitter:description" content="Book for for the ICME series courses" />
  

<meta name="author" content="Manuela Coci" />
<meta name="author" content="Luigi Gallucci" />
<meta name="author" content="Davide Corso" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="metagenomics.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<div style="text-align:center">
<li class="toc-logo"><a href="./"><img src="img/icme_13.png" width="250" height="250"></a></li>
<li><a href="https://www.microbeco.org/icme13-roma-2024/"><strong>ICME 13</strong></a></li>
</div>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#icme-13---rome"><i class="fa fa-check"></i><b>1.1</b> ICME 13 - Rome</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>2</b> Introduction to the command line</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#set-up-a-terminal"><i class="fa fa-check"></i><b>2.1</b> Set-up a terminal</a></li>
<li class="chapter" data-level="2.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#working-with-the-command-line"><i class="fa fa-check"></i><b>2.2</b> Working with the command line</a></li>
<li class="chapter" data-level="2.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#playing-around-with-basic-unix-commands"><i class="fa fa-check"></i><b>2.3</b> Playing around with basic UNIX commands</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#some-notes"><i class="fa fa-check"></i><b>2.3.1</b> Some notes!</a></li>
<li class="chapter" data-level="2.3.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-and-navigating-directories"><i class="fa fa-check"></i><b>2.3.2</b> Creating and navigating directories</a></li>
<li class="chapter" data-level="2.3.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-a-new-file"><i class="fa fa-check"></i><b>2.3.3</b> Creating a new file</a></li>
<li class="chapter" data-level="2.3.4" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#copying-renaming-moving-and-deleting-files"><i class="fa fa-check"></i><b>2.3.4</b> Copying, renaming, moving and deleting files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="s-analysis.html"><a href="s-analysis.html"><i class="fa fa-check"></i><b>3</b> 16s Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="s-analysis.html"><a href="s-analysis.html#commands-and-features-that-we-will-use-in-our-practice"><i class="fa fa-check"></i><b>3.1</b> Commands and features that we will use in our practice</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="s-analysis.html"><a href="s-analysis.html#touch"><i class="fa fa-check"></i><b>3.1.1</b> <code>touch</code></a></li>
<li class="chapter" data-level="3.1.2" data-path="s-analysis.html"><a href="s-analysis.html#echo"><i class="fa fa-check"></i><b>3.1.2</b> <code>echo</code></a></li>
<li class="chapter" data-level="3.1.3" data-path="s-analysis.html"><a href="s-analysis.html#cat"><i class="fa fa-check"></i><b>3.1.3</b> <code>cat</code></a></li>
<li class="chapter" data-level="3.1.4" data-path="s-analysis.html"><a href="s-analysis.html#section"><i class="fa fa-check"></i><b>3.1.4</b> <code>$</code></a></li>
<li class="chapter" data-level="3.1.5" data-path="s-analysis.html"><a href="s-analysis.html#chmod"><i class="fa fa-check"></i><b>3.1.5</b> <code>chmod</code></a></li>
<li class="chapter" data-level="3.1.6" data-path="s-analysis.html"><a href="s-analysis.html#basename"><i class="fa fa-check"></i><b>3.1.6</b> <code>basename</code></a></li>
<li class="chapter" data-level="3.1.7" data-path="s-analysis.html"><a href="s-analysis.html#create-a-shell-scripts"><i class="fa fa-check"></i><b>3.1.7</b> Create a shell scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="s-analysis.html"><a href="s-analysis.html#package-manager"><i class="fa fa-check"></i><b>3.2</b> Package Manager</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="s-analysis.html"><a href="s-analysis.html#conda"><i class="fa fa-check"></i><b>3.2.1</b> Conda</a></li>
<li class="chapter" data-level="3.2.2" data-path="s-analysis.html"><a href="s-analysis.html#mamba-recommended"><i class="fa fa-check"></i><b>3.2.2</b> Mamba (Recommended)</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="s-analysis.html"><a href="s-analysis.html#base-calling"><i class="fa fa-check"></i><b>3.3</b> Base Calling</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="s-analysis.html"><a href="s-analysis.html#guppy"><i class="fa fa-check"></i><b>3.3.1</b> Guppy</a></li>
<li class="chapter" data-level="3.3.2" data-path="s-analysis.html"><a href="s-analysis.html#dorado"><i class="fa fa-check"></i><b>3.3.2</b> Dorado</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="s-analysis.html"><a href="s-analysis.html#filtering-and-trimming"><i class="fa fa-check"></i><b>3.4</b> Filtering and Trimming</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="s-analysis.html"><a href="s-analysis.html#nanofilt"><i class="fa fa-check"></i><b>3.4.1</b> NanoFilt</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="s-analysis.html"><a href="s-analysis.html#subsetting"><i class="fa fa-check"></i><b>3.5</b> Subsetting</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="s-analysis.html"><a href="s-analysis.html#bbmap-tools"><i class="fa fa-check"></i><b>3.5.1</b> BBMAP Tools</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="s-analysis.html"><a href="s-analysis.html#taxonomic-assignment"><i class="fa fa-check"></i><b>3.6</b> Taxonomic Assignment</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="s-analysis.html"><a href="s-analysis.html#emu"><i class="fa fa-check"></i><b>3.6.1</b> EMU</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="metagenomics.html"><a href="metagenomics.html"><i class="fa fa-check"></i><b>4</b> Metagenomics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="metagenomics.html"><a href="metagenomics.html#reads-quality-check"><i class="fa fa-check"></i><b>4.1</b> Reads Quality Check</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="metagenomics.html"><a href="metagenomics.html#bbmap-stats.sh"><i class="fa fa-check"></i><b>4.1.1</b> BBMap / <code>stats.sh</code></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="metagenomics.html"><a href="metagenomics.html#assembly"><i class="fa fa-check"></i><b>4.2</b> Assembly</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="metagenomics.html"><a href="metagenomics.html#flye-metaflye"><i class="fa fa-check"></i><b>4.2.1</b> Flye (Metaflye)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="metagenomics.html"><a href="metagenomics.html#read-mapping"><i class="fa fa-check"></i><b>4.3</b> Read mapping</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="metagenomics.html"><a href="metagenomics.html#minimap2"><i class="fa fa-check"></i><b>4.3.1</b> Minimap2</a></li>
<li class="chapter" data-level="4.3.2" data-path="metagenomics.html"><a href="metagenomics.html#samtools"><i class="fa fa-check"></i><b>4.3.2</b> samtools</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="metagenomics.html"><a href="metagenomics.html#binning"><i class="fa fa-check"></i><b>4.4</b> Binning</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="metagenomics.html"><a href="metagenomics.html#semibin2"><i class="fa fa-check"></i><b>4.4.1</b> SemiBin2</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="metagenomics.html"><a href="metagenomics.html#mags-quality-check"><i class="fa fa-check"></i><b>4.5</b> MAGs Quality Check</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="metagenomics.html"><a href="metagenomics.html#checkm2"><i class="fa fa-check"></i><b>4.5.1</b> CheckM2</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="metagenomics.html"><a href="metagenomics.html#taxonomic-classification"><i class="fa fa-check"></i><b>4.6</b> Taxonomic Classification</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="metagenomics.html"><a href="metagenomics.html#gtdb-tk"><i class="fa fa-check"></i><b>4.6.1</b> GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="metagenomics.html"><a href="metagenomics.html#functional-annotation"><i class="fa fa-check"></i><b>4.7</b> Functional Annotation</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="metagenomics.html"><a href="metagenomics.html#anvio"><i class="fa fa-check"></i><b>4.7.1</b> Anvi’o</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html"><i class="fa fa-check"></i><b>5</b> Our Metagenomic practice</a>
<ul>
<li class="chapter" data-level="5.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#orf-prediction"><i class="fa fa-check"></i><b>5.1</b> ORF Prediction</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#prodigal"><i class="fa fa-check"></i><b>5.1.1</b> Prodigal</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#hidden-markov-model"><i class="fa fa-check"></i><b>5.2</b> Hidden Markov Model</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#hmmer"><i class="fa fa-check"></i><b>5.2.1</b> hmmer</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#assembly-quality-check"><i class="fa fa-check"></i><b>5.3</b> Assembly Quality Check</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#quast"><i class="fa fa-check"></i><b>5.3.1</b> Quast</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#web-tools-annotations"><i class="fa fa-check"></i><b>5.4</b> Web Tools Annotations</a></li>
<li class="chapter" data-level="5.5" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#functional-annotation---mags-based"><i class="fa fa-check"></i><b>5.5</b> Functional Annotation - MAGs based</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ICME 13 - Rome Edition</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="our-metagenomic-practice" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Our Metagenomic practice<a href="our-metagenomic-practice.html#our-metagenomic-practice" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In our practice we will cover the functional annotation on metagenomics data, both on assembly based and MAGs based. Due to resources and time limits we will try to run Prodigal, HMMer and Quast.</p>
<p>For the assembly-based analysis, we have the assembly fasta file located in the following path: <code>/SERVER/mg_data/Assembly_metaflye/assembly.fasta</code></p>
<div id="orf-prediction" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> ORF Prediction<a href="our-metagenomic-practice.html#orf-prediction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Fast, reliable protein-coding gene prediction for prokaryotic genomes.</p>
<p><a href="https://github.com/hyattpd/Prodigal">https://github.com/hyattpd/Prodigal</a></p>
<div id="prodigal" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Prodigal<a href="our-metagenomic-practice.html#prodigal" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Setup the conda env</strong></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb52-1"><a href="our-metagenomic-practice.html#cb52-1" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> prodigal <span class="at">-c</span> bioconda <span class="at">-c</span> conda-forge prodigal</span></code></pre></div>
<p>Tips:</p>
<ul>
<li>Input: Prodigal run using a fasta file, for example the one represented the assembly</li>
<li>Output: You should obtains a GFF file and an Aminoacidic Fasta file of the predicted orfs</li>
</ul>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s01_prodigal.sh</code></p>
<p><code>touch s01_prodigal.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb53-1"><a href="our-metagenomic-practice.html#cb53-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb53-2"><a href="our-metagenomic-practice.html#cb53-2" tabindex="-1"></a></span>
<span id="cb53-3"><a href="our-metagenomic-practice.html#cb53-3" tabindex="-1"></a><span class="va">assembly</span><span class="op">=</span><span class="st">&quot;/SERVER/mg_data/Assembly_metaflye/assembly.fasta&quot;</span></span>
<span id="cb53-4"><a href="our-metagenomic-practice.html#cb53-4" tabindex="-1"></a></span>
<span id="cb53-5"><a href="our-metagenomic-practice.html#cb53-5" tabindex="-1"></a></span>
<span id="cb53-6"><a href="our-metagenomic-practice.html#cb53-6" tabindex="-1"></a><span class="va">outfolder</span><span class="op">=</span><span class="st">&quot;output_s01&quot;</span></span>
<span id="cb53-7"><a href="our-metagenomic-practice.html#cb53-7" tabindex="-1"></a></span>
<span id="cb53-8"><a href="our-metagenomic-practice.html#cb53-8" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">$outfolder</span></span>
<span id="cb53-9"><a href="our-metagenomic-practice.html#cb53-9" tabindex="-1"></a></span>
<span id="cb53-10"><a href="our-metagenomic-practice.html#cb53-10" tabindex="-1"></a><span class="ex">prodigal</span> <span class="at">-i</span> <span class="va">${assembly}</span> <span class="dt">\</span></span>
<span id="cb53-11"><a href="our-metagenomic-practice.html#cb53-11" tabindex="-1"></a>    <span class="at">-o</span> <span class="va">${outfolder}</span>/genes.gff <span class="dt">\</span></span>
<span id="cb53-12"><a href="our-metagenomic-practice.html#cb53-12" tabindex="-1"></a>    <span class="at">-a</span> <span class="va">${outfolder}</span>/protein_translations.faa <span class="dt">\</span></span>
<span id="cb53-13"><a href="our-metagenomic-practice.html#cb53-13" tabindex="-1"></a>    <span class="at">-f</span> gff <span class="dt">\</span></span>
<span id="cb53-14"><a href="our-metagenomic-practice.html#cb53-14" tabindex="-1"></a>    <span class="at">-p</span> meta</span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_MG/output_s01/</code></pre>
<p>Change its permission:
<code>chmod u+x s01_prodigal.sh</code></p>
<p>Execute it:
<code>./s01_prodigal.sh</code></p>
</details>
</div>
</div>
<div id="hidden-markov-model" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Hidden Markov Model<a href="our-metagenomic-practice.html#hidden-markov-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>HMMER is a software package that provides tools for making probabilistic models of protein and DNA sequence domain families – called profile hidden Markov models, profile HMMs, or just profiles.</p>
<p>HMMER is used for searching sequence databases for sequence homologs, and for making sequence alignments. It implements methods using probabilistic models called profile hidden Markov models (profile HMMs).</p>
<p><a href="https://github.com/EBI-Metagenomics/hmmer3">https://github.com/EBI-Metagenomics/hmmer3</a></p>
<div id="hmmer" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> hmmer<a href="our-metagenomic-practice.html#hmmer" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Setup the conda env</strong></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb55-1"><a href="our-metagenomic-practice.html#cb55-1" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> hmmer <span class="at">-c</span> bioconda <span class="at">-c</span> conda-forge hmmer</span></code></pre></div>
<div id="split-pfam-files" class="section level4 hasAnchor" number="5.2.1.1">
<h4><span class="header-section-number">5.2.1.1</span> Split PFAM files<a href="our-metagenomic-practice.html#split-pfam-files" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Pfam is a comprehensive collection of protein domains and families, represented as multiple sequence alignments and as profile hidden Markov models.</p>
<p>Download the file <code>Pfam-A.hmm.gz</code> from the PFAM FTP Server: <a href="https://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/">https://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/</a></p>
<div id="step-0" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 0</strong><a href="our-metagenomic-practice.html#step-0" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>De-Compress the file <code>Pfam-A.hmm.gz</code>. How will you do it?
TIPS:</p>
<ul>
<li>Search on google</li>
</ul>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p><code>gzip -d Pfam-A.hmm.gz</code></p>
</details>
</div>
<div id="step-1" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 1</strong><a href="our-metagenomic-practice.html#step-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Extract the name of the models in the PFAM file <strong>(Advanced Task)</strong></p>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="our-metagenomic-practice.html#cb56-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;^NAME&quot;</span> Pfam-A.hmm <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $2}&#39;</span> <span class="op">&gt;</span> all_names.txt</span></code></pre></div>
</details>
</div>
<div id="step-2" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 2</strong><a href="our-metagenomic-practice.html#step-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Using the retrieved names, extract the corresponding models into multiple files.</p>
<p>Command to use: <code>hmmfetch</code> from the hmmer tools</p>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s02_splitPFAM.sh</code></p>
<p><code>touch s02_splitPFAM.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb57-1"><a href="our-metagenomic-practice.html#cb57-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb57-2"><a href="our-metagenomic-practice.html#cb57-2" tabindex="-1"></a></span>
<span id="cb57-3"><a href="our-metagenomic-practice.html#cb57-3" tabindex="-1"></a><span class="va">outfolder</span><span class="op">=</span><span class="st">&quot;output_s02&quot;</span></span>
<span id="cb57-4"><a href="our-metagenomic-practice.html#cb57-4" tabindex="-1"></a></span>
<span id="cb57-5"><a href="our-metagenomic-practice.html#cb57-5" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">name</span></span>
<span id="cb57-6"><a href="our-metagenomic-practice.html#cb57-6" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb57-7"><a href="our-metagenomic-practice.html#cb57-7" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$name</span><span class="st">&quot;</span></span>
<span id="cb57-8"><a href="our-metagenomic-practice.html#cb57-8" tabindex="-1"></a>    <span class="ex">hmmfetch</span> Pfam-A.hmm <span class="va">$name</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${outfolder}</span><span class="st">/</span><span class="va">$name</span><span class="st">.hmm&quot;</span></span>
<span id="cb57-9"><a href="our-metagenomic-practice.html#cb57-9" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> 500_names.txt</span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_MG/output_s02/</code></pre>
<p>Change its permission:
<code>chmod u+x s02_splitPFAM.sh</code></p>
<p>Execute it:
<code>./s02_splitPFAM.sh</code></p>
</details>
</div>
</div>
<div id="hmmsearch" class="section level4 hasAnchor" number="5.2.1.2">
<h4><span class="header-section-number">5.2.1.2</span> <code>hmmsearch</code><a href="our-metagenomic-practice.html#hmmsearch" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now execute <code>hmmsearch</code> to the retrieved ORFs to annotate them based on the extracted models.</p>
<p>If you are working on the CNR system, use only 1 cpu.</p>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s03_hmmsearch.sh</code></p>
<p><code>touch s03_hmmsearch.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb59-1"><a href="our-metagenomic-practice.html#cb59-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb59-2"><a href="our-metagenomic-practice.html#cb59-2" tabindex="-1"></a></span>
<span id="cb59-3"><a href="our-metagenomic-practice.html#cb59-3" tabindex="-1"></a></span>
<span id="cb59-4"><a href="our-metagenomic-practice.html#cb59-4" tabindex="-1"></a><span class="va">outfolder</span><span class="op">=</span><span class="st">&quot;output_s03&quot;</span></span>
<span id="cb59-5"><a href="our-metagenomic-practice.html#cb59-5" tabindex="-1"></a></span>
<span id="cb59-6"><a href="our-metagenomic-practice.html#cb59-6" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">name</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb59-7"><a href="our-metagenomic-practice.html#cb59-7" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$name</span><span class="st">&quot;</span></span>
<span id="cb59-8"><a href="our-metagenomic-practice.html#cb59-8" tabindex="-1"></a></span>
<span id="cb59-9"><a href="our-metagenomic-practice.html#cb59-9" tabindex="-1"></a>    <span class="va">model_folder</span><span class="op">=</span><span class="st">&quot;</span><span class="va">${outfolder}</span><span class="st">/</span><span class="va">${name}</span><span class="st">&quot;</span></span>
<span id="cb59-10"><a href="our-metagenomic-practice.html#cb59-10" tabindex="-1"></a>    <span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${model_folder}</span></span>
<span id="cb59-11"><a href="our-metagenomic-practice.html#cb59-11" tabindex="-1"></a></span>
<span id="cb59-12"><a href="our-metagenomic-practice.html#cb59-12" tabindex="-1"></a>    <span class="va">model</span><span class="op">=</span><span class="st">&quot;output_s02/</span><span class="va">${name}</span><span class="st">.hmm&quot;</span></span>
<span id="cb59-13"><a href="our-metagenomic-practice.html#cb59-13" tabindex="-1"></a></span>
<span id="cb59-14"><a href="our-metagenomic-practice.html#cb59-14" tabindex="-1"></a>    <span class="ex">hmmsearch</span> <span class="at">--tblout</span> <span class="va">${model_folder}</span>/table.out <span class="at">-o</span> <span class="va">${model_folder}</span>/align.out <span class="at">-E</span> 0.000001 <span class="at">--cpu</span> 1 <span class="at">--notextw</span> <span class="va">${model}</span> output_s01/protein_translations.faa</span>
<span id="cb59-15"><a href="our-metagenomic-practice.html#cb59-15" tabindex="-1"></a></span>
<span id="cb59-16"><a href="our-metagenomic-practice.html#cb59-16" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> 500_names.txt</span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_MG/output_s03/</code></pre>
<p>Change its permission:
<code>chmod u+x s03_hmmsearch.sh</code></p>
<p>Execute it:
<code>./s03_hmmsearch.sh</code></p>
</details>
</div>
</div>
</div>
<div id="assembly-quality-check" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Assembly Quality Check<a href="our-metagenomic-practice.html#assembly-quality-check" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="quast" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Quast<a href="our-metagenomic-practice.html#quast" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The QUAST package works both with and without reference genomes. However, it is much more informative if at least a close reference genome is provided along with the assemblies. The tool accepts multiple assemblies, thus is suitable for comparison.</p>
<p><a href="https://github.com/ablab/quast">https://github.com/ablab/quast</a></p>
<p><strong>Setup the conda env</strong></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb61-1"><a href="our-metagenomic-practice.html#cb61-1" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> quast <span class="at">-c</span> bioconda <span class="at">-c</span> conda-forge quast</span></code></pre></div>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s04_quast.sh</code></p>
<p><code>touch s04_quast.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="our-metagenomic-practice.html#cb62-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb62-2"><a href="our-metagenomic-practice.html#cb62-2" tabindex="-1"></a></span>
<span id="cb62-3"><a href="our-metagenomic-practice.html#cb62-3" tabindex="-1"></a><span class="va">outfolder</span><span class="op">=</span><span class="st">&quot;output_s04&quot;</span></span>
<span id="cb62-4"><a href="our-metagenomic-practice.html#cb62-4" tabindex="-1"></a></span>
<span id="cb62-5"><a href="our-metagenomic-practice.html#cb62-5" tabindex="-1"></a></span>
<span id="cb62-6"><a href="our-metagenomic-practice.html#cb62-6" tabindex="-1"></a><span class="ex">quast</span> <span class="at">--labels</span> flye <span class="at">--contig-thresholds</span> 0,1000,10000,100000,1000000 <span class="at">--threads</span> 2 <span class="at">-o</span> <span class="va">${outfolder}</span> /SERVER/mg_data/Assembly_metaflye/assembly.fasta</span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_MG/output_s04/</code></pre>
<p>Change its permission:
<code>chmod u+x s04_quast.sh</code></p>
<p>Execute it:
<code>./s04_quast.sh</code></p>
</details>
<p>What if you use <code>metaquast</code>?</p>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>MetaQUAST the extension for metagenomic datasets, it evaluates and compares metagenome assemblies based on alignments to close references. It is based on QUAST genome quality assessment tool, but addresses features specific for metagenome datasets.</p>
</details>
</div>
</div>
<div id="web-tools-annotations" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Web Tools Annotations<a href="our-metagenomic-practice.html#web-tools-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>KOFAM Koala <a href="https://www.genome.jp/tools/kofamkoala/">https://www.genome.jp/tools/kofamkoala/</a></li>
<li>EGGNOG <a href="http://eggnog-mapper.embl.de">http://eggnog-mapper.embl.de</a></li>
<li>DBcan <a href="https://bcb.unl.edu/dbCAN2/blast.php">https://bcb.unl.edu/dbCAN2/blast.php</a></li>
</ul>
</div>
<div id="functional-annotation---mags-based" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Functional Annotation - MAGs based<a href="our-metagenomic-practice.html#functional-annotation---mags-based" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you have finished, try to run the same analysis on each MAGs located in:
<code>/SERVER/mg_data/mg/genomi_per_annotazione</code></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="metagenomics.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["icme13.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
