<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 16s Analysis | ICME 13 - Rome Edition</title>
  <meta name="description" content="Book for for the ICME series courses" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="3 16s Analysis | ICME 13 - Rome Edition" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Book for for the ICME series courses" />
  <meta name="github-repo" content="microbeco/microbeco.github.io" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 16s Analysis | ICME 13 - Rome Edition" />
  
  <meta name="twitter:description" content="Book for for the ICME series courses" />
  

<meta name="author" content="Manuela Coci" />
<meta name="author" content="Luigi Gallucci" />
<meta name="author" content="Davide Corso" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction-to-the-command-line.html"/>
<link rel="next" href="metagenomics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<div style="text-align:center">
<li class="toc-logo"><a href="./"><img src="img/icme_13.png" width="250" height="250"></a></li>
<li><a href="https://www.microbeco.org/icme13-roma-2024/"><strong>ICME 13</strong></a></li>
</div>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#icme-13---rome"><i class="fa fa-check"></i><b>1.1</b> ICME 13 - Rome</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html"><i class="fa fa-check"></i><b>2</b> Introduction to the command line</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#set-up-a-terminal"><i class="fa fa-check"></i><b>2.1</b> Set-up a terminal</a></li>
<li class="chapter" data-level="2.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#working-with-the-command-line"><i class="fa fa-check"></i><b>2.2</b> Working with the command line</a></li>
<li class="chapter" data-level="2.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#playing-around-with-basic-unix-commands"><i class="fa fa-check"></i><b>2.3</b> Playing around with basic UNIX commands</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#some-notes"><i class="fa fa-check"></i><b>2.3.1</b> Some notes!</a></li>
<li class="chapter" data-level="2.3.2" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-and-navigating-directories"><i class="fa fa-check"></i><b>2.3.2</b> Creating and navigating directories</a></li>
<li class="chapter" data-level="2.3.3" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#creating-a-new-file"><i class="fa fa-check"></i><b>2.3.3</b> Creating a new file</a></li>
<li class="chapter" data-level="2.3.4" data-path="introduction-to-the-command-line.html"><a href="introduction-to-the-command-line.html#copying-renaming-moving-and-deleting-files"><i class="fa fa-check"></i><b>2.3.4</b> Copying, renaming, moving and deleting files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="s-analysis.html"><a href="s-analysis.html"><i class="fa fa-check"></i><b>3</b> 16s Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="s-analysis.html"><a href="s-analysis.html#commands-and-features-that-we-will-use-in-our-practice"><i class="fa fa-check"></i><b>3.1</b> Commands and features that we will use in our practice</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="s-analysis.html"><a href="s-analysis.html#touch"><i class="fa fa-check"></i><b>3.1.1</b> <code>touch</code></a></li>
<li class="chapter" data-level="3.1.2" data-path="s-analysis.html"><a href="s-analysis.html#echo"><i class="fa fa-check"></i><b>3.1.2</b> <code>echo</code></a></li>
<li class="chapter" data-level="3.1.3" data-path="s-analysis.html"><a href="s-analysis.html#cat"><i class="fa fa-check"></i><b>3.1.3</b> <code>cat</code></a></li>
<li class="chapter" data-level="3.1.4" data-path="s-analysis.html"><a href="s-analysis.html#section"><i class="fa fa-check"></i><b>3.1.4</b> <code>$</code></a></li>
<li class="chapter" data-level="3.1.5" data-path="s-analysis.html"><a href="s-analysis.html#chmod"><i class="fa fa-check"></i><b>3.1.5</b> <code>chmod</code></a></li>
<li class="chapter" data-level="3.1.6" data-path="s-analysis.html"><a href="s-analysis.html#basename"><i class="fa fa-check"></i><b>3.1.6</b> <code>basename</code></a></li>
<li class="chapter" data-level="3.1.7" data-path="s-analysis.html"><a href="s-analysis.html#create-a-shell-scripts"><i class="fa fa-check"></i><b>3.1.7</b> Create a shell scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="s-analysis.html"><a href="s-analysis.html#package-manager"><i class="fa fa-check"></i><b>3.2</b> Package Manager</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="s-analysis.html"><a href="s-analysis.html#conda"><i class="fa fa-check"></i><b>3.2.1</b> Conda</a></li>
<li class="chapter" data-level="3.2.2" data-path="s-analysis.html"><a href="s-analysis.html#mamba-recommended"><i class="fa fa-check"></i><b>3.2.2</b> Mamba (Recommended)</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="s-analysis.html"><a href="s-analysis.html#base-calling"><i class="fa fa-check"></i><b>3.3</b> Base Calling</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="s-analysis.html"><a href="s-analysis.html#guppy"><i class="fa fa-check"></i><b>3.3.1</b> Guppy</a></li>
<li class="chapter" data-level="3.3.2" data-path="s-analysis.html"><a href="s-analysis.html#dorado"><i class="fa fa-check"></i><b>3.3.2</b> Dorado</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="s-analysis.html"><a href="s-analysis.html#filtering-and-trimming"><i class="fa fa-check"></i><b>3.4</b> Filtering and Trimming</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="s-analysis.html"><a href="s-analysis.html#nanofilt"><i class="fa fa-check"></i><b>3.4.1</b> NanoFilt</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="s-analysis.html"><a href="s-analysis.html#subsetting"><i class="fa fa-check"></i><b>3.5</b> Subsetting</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="s-analysis.html"><a href="s-analysis.html#bbmap-tools"><i class="fa fa-check"></i><b>3.5.1</b> BBMAP Tools</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="s-analysis.html"><a href="s-analysis.html#taxonomic-assignment"><i class="fa fa-check"></i><b>3.6</b> Taxonomic Assignment</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="s-analysis.html"><a href="s-analysis.html#emu"><i class="fa fa-check"></i><b>3.6.1</b> EMU</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="metagenomics.html"><a href="metagenomics.html"><i class="fa fa-check"></i><b>4</b> Metagenomics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="metagenomics.html"><a href="metagenomics.html#reads-quality-check"><i class="fa fa-check"></i><b>4.1</b> Reads Quality Check</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="metagenomics.html"><a href="metagenomics.html#bbmap-stats.sh"><i class="fa fa-check"></i><b>4.1.1</b> BBMap / <code>stats.sh</code></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="metagenomics.html"><a href="metagenomics.html#assembly"><i class="fa fa-check"></i><b>4.2</b> Assembly</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="metagenomics.html"><a href="metagenomics.html#flye-metaflye"><i class="fa fa-check"></i><b>4.2.1</b> Flye (Metaflye)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="metagenomics.html"><a href="metagenomics.html#read-mapping"><i class="fa fa-check"></i><b>4.3</b> Read mapping</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="metagenomics.html"><a href="metagenomics.html#minimap2"><i class="fa fa-check"></i><b>4.3.1</b> Minimap2</a></li>
<li class="chapter" data-level="4.3.2" data-path="metagenomics.html"><a href="metagenomics.html#samtools"><i class="fa fa-check"></i><b>4.3.2</b> samtools</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="metagenomics.html"><a href="metagenomics.html#binning"><i class="fa fa-check"></i><b>4.4</b> Binning</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="metagenomics.html"><a href="metagenomics.html#semibin2"><i class="fa fa-check"></i><b>4.4.1</b> SemiBin2</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="metagenomics.html"><a href="metagenomics.html#mags-quality-check"><i class="fa fa-check"></i><b>4.5</b> MAGs Quality Check</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="metagenomics.html"><a href="metagenomics.html#checkm2"><i class="fa fa-check"></i><b>4.5.1</b> CheckM2</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="metagenomics.html"><a href="metagenomics.html#taxonomic-classification"><i class="fa fa-check"></i><b>4.6</b> Taxonomic Classification</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="metagenomics.html"><a href="metagenomics.html#gtdb-tk"><i class="fa fa-check"></i><b>4.6.1</b> GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="metagenomics.html"><a href="metagenomics.html#functional-annotation"><i class="fa fa-check"></i><b>4.7</b> Functional Annotation</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="metagenomics.html"><a href="metagenomics.html#anvio"><i class="fa fa-check"></i><b>4.7.1</b> Anvi’o</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html"><i class="fa fa-check"></i><b>5</b> Our Metagenomic practice</a>
<ul>
<li class="chapter" data-level="5.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#orf-prediction"><i class="fa fa-check"></i><b>5.1</b> ORF Prediction</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#prodigal"><i class="fa fa-check"></i><b>5.1.1</b> Prodigal</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#hidden-markov-model"><i class="fa fa-check"></i><b>5.2</b> Hidden Markov Model</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#hmmer"><i class="fa fa-check"></i><b>5.2.1</b> hmmer</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#assembly-quality-check"><i class="fa fa-check"></i><b>5.3</b> Assembly Quality Check</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#quast"><i class="fa fa-check"></i><b>5.3.1</b> Quast</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#web-tools-annotations"><i class="fa fa-check"></i><b>5.4</b> Web Tools Annotations</a></li>
<li class="chapter" data-level="5.5" data-path="our-metagenomic-practice.html"><a href="our-metagenomic-practice.html#functional-annotation---mags-based"><i class="fa fa-check"></i><b>5.5</b> Functional Annotation - MAGs based</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ICME 13 - Rome Edition</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="s-analysis" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> 16s Analysis<a href="s-analysis.html#s-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this tutorial we are going to see/use different commands and tools to perform the 16s analysis on long reads NanoPore. However, the practical analysis will cover only some of these steps as data has been prepared in advance due to their computation and time consuming limits.</p>
<p><strong>This guide has been created with the purpose of a practical crush course and it is not intended as a complete reference but rather as a beginners pipeline to analyze NanoPore generated data.</strong></p>
<div id="commands-and-features-that-we-will-use-in-our-practice" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Commands and features that we will use in our practice<a href="s-analysis.html#commands-and-features-that-we-will-use-in-our-practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="touch" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> <code>touch</code><a href="s-analysis.html#touch" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create a new empty file.</p>
<p>Example: <code>touch prova.txt</code></p>
</div>
<div id="echo" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> <code>echo</code><a href="s-analysis.html#echo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The echo command is a built-in Linux feature that prints out arguments as the standard output. echo is commonly used <em>to display text strings or command results as messages</em>.</p>
<p>Example: <code>echo "Hello World"</code></p>
</div>
<div id="cat" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> <code>cat</code><a href="s-analysis.html#cat" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Concatenate or print the contents of a file</p>
<p>Example: <code>cat prova.txt</code></p>
</div>
<div id="section" class="section level3 hasAnchor" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> <code>$</code><a href="s-analysis.html#section" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Append <code>$</code> to the variable name to access the variable value.</p>
<p>Example:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="s-analysis.html#cb21-1" tabindex="-1"></a><span class="va">var</span><span class="op">=</span><span class="st">&quot;Hello World&quot;</span></span>
<span id="cb21-2"><a href="s-analysis.html#cb21-2" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$var</span></span></code></pre></div>
<p>Output:</p>
<blockquote>
<p>Hello World</p>
</blockquote>
</div>
<div id="chmod" class="section level3 hasAnchor" number="3.1.5">
<h3><span class="header-section-number">3.1.5</span> <code>chmod</code><a href="s-analysis.html#chmod" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Change the permissions of a file or directory and make it executable.</p>
<ul>
<li>use the “chmod +x” command on a system file to give permission to all users to execute it.</li>
<li>use the “chmod u+x” for made the file executable for your user.</li>
</ul>
<p>Example: <code>chmod u+x s01_filtering.sh</code></p>
</div>
<div id="basename" class="section level3 hasAnchor" number="3.1.6">
<h3><span class="header-section-number">3.1.6</span> <code>basename</code><a href="s-analysis.html#basename" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It removes the path from a file string, providing only its filename and trailing suffix from given file names.</p>
<p>Example:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="s-analysis.html#cb22-1" tabindex="-1"></a><span class="fu">basename</span> /path/to/filename.txt</span></code></pre></div>
<p>Output:</p>
<blockquote>
<p>filename.txt</p>
</blockquote>
</div>
<div id="create-a-shell-scripts" class="section level3 hasAnchor" number="3.1.7">
<h3><span class="header-section-number">3.1.7</span> Create a shell scripts<a href="s-analysis.html#create-a-shell-scripts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Sometime you don’t need to run a command at a time, we can pre-think, organize series of actions (a program) that you can then execute within Bash.</p>
<p>For example we can write a shell script that runs a series of commands and we can run the script from the terminal to execute all the steps that we have integrated in the script.</p>
<p>For example, lets assume that we want to visualize the first four reads from a FASTQ and redirect to a file. For this task we are going to create an empty file and write in it the following text:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="s-analysis.html#cb23-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb23-2"><a href="s-analysis.html#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="s-analysis.html#cb23-3" tabindex="-1"></a><span class="fu">cat</span> /SERVER/16s_data/mysample.FASTQ <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 4 <span class="op">&gt;</span> first_read.fastq</span>
<span id="cb23-4"><a href="s-analysis.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="s-analysis.html#cb23-5" tabindex="-1"></a><span class="va">var</span><span class="op">=</span><span class="st">&quot;Hello World&quot;</span></span>
<span id="cb23-6"><a href="s-analysis.html#cb23-6" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$var</span></span></code></pre></div>
<p>Now we give the ‘executable’ permission to our script in order to be executed:</p>
<p><code>chmod u+x myscript.sh</code></p>
<p>Now we execute our script:</p>
<p><code>./myscript.sh</code></p>
<ul>
<li>The first row must be <code>#!/bin/bash</code> to allows the shell to interpret your code with bash</li>
<li>The symbol <code>|</code> is the PIPE, it lets you connect actions: the output of a command is the input of the next command</li>
<li>The command <code>head</code> allows to print a specified number of rows from an input.</li>
<li>Sometime different actions cannot be linked with a PIPE, in this case we use to write the series of actions in each line, as we did in the last two rows of our scripts.</li>
</ul>
</div>
</div>
<div id="package-manager" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Package Manager<a href="s-analysis.html#package-manager" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="conda" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Conda<a href="s-analysis.html#conda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Conda is a powerful command line tool for package and environment management that runs on Windows, macOS, and Linux.</p>
<p><a href="https://conda.io/projects/conda/en/latest/user-guide/getting-started.html">https://conda.io/projects/conda/en/latest/user-guide/getting-started.html</a></p>
<p>Within conda, you can create, export, list, remove, and update environments that have different versions of Python and/or packages installed in them. <em>Switching or moving between environments is called activating the environment</em>. You can also share an environment file.</p>
</div>
<div id="mamba-recommended" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Mamba (Recommended)<a href="s-analysis.html#mamba-recommended" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Mamba is a reimplementation of the conda package manager in C++, so it is faster and more convenient due to its faster dependencies solving.</p>
<p><a href="https://mamba.readthedocs.io/en/latest/user_guide/mamba.html">https://mamba.readthedocs.io/en/latest/user_guide/mamba.html</a></p>
<div id="creating-a-condamamba-env" class="section level4 hasAnchor" number="3.2.2.1">
<h4><span class="header-section-number">3.2.2.1</span> Creating a conda/mamba env<a href="s-analysis.html#creating-a-condamamba-env" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To create an environment:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="s-analysis.html#cb24-1" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> myenv</span></code></pre></div>
<p>To activate a created environment:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="s-analysis.html#cb25-1" tabindex="-1"></a><span class="ex">conda</span> activate myenv</span></code></pre></div>
</div>
<div id="installing-tools" class="section level4 hasAnchor" number="3.2.2.2">
<h4><span class="header-section-number">3.2.2.2</span> Installing tools<a href="s-analysis.html#installing-tools" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Most of the bioinformatics packages can be searched in Bioconda at this link <a href="https://bioconda.github.io">https://bioconda.github.io</a>, but also in <code>conda-forge</code> channel and then we can install them by executing</p>
<pre><code>mamba install -c bioconda -c conda-forge nanofilt</code></pre>
<p>For our practical analysis, conda environments have already been created.</p>
<p>The following environment were created:</p>
<ul>
<li><code>mamba activate /home/irsa/miniconda3/envs/ONTpp</code></li>
<li><code>mamba activate /home/irsa/miniconda3/envs/emu</code></li>
</ul>
</div>
</div>
</div>
<div id="base-calling" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Base Calling<a href="s-analysis.html#base-calling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Base calling is the process of translating the electronic raw signal of the sequencer into bases, i.e., ATCG and converting the raw files (FAST5) to a FASTQ files (human-readable), which contains the nucleotide sequences of the reads.</p>
<p>Raw data are huge in terms of storage, and since basecalling is computationally and time demanding, the fastq files are already provided.</p>
<p>However, to perform this step we suggest to use one of the two following tools:</p>
<div id="guppy" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Guppy<a href="s-analysis.html#guppy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://community.nanoporetech.com/docs/prepare/library_prep_protocols/Guppy-protocol/v/gpb_2003_v1_revax_14dec2018/guppy-software-overview">https://community.nanoporetech.com/docs/prepare/library_prep_protocols/Guppy-protocol/v/gpb_2003_v1_revax_14dec2018/guppy-software-overview</a></p>
<p>Guppy usage:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="s-analysis.html#cb27-1" tabindex="-1"></a><span class="ex">guppy_basecaller</span> <span class="dt">\</span></span>
<span id="cb27-2"><a href="s-analysis.html#cb27-2" tabindex="-1"></a>  <span class="at">--num_callers</span> 4 <span class="dt">\</span></span>
<span id="cb27-3"><a href="s-analysis.html#cb27-3" tabindex="-1"></a>  <span class="at">--cpu_threads_per_caller</span> 64 <span class="dt">\</span></span>
<span id="cb27-4"><a href="s-analysis.html#cb27-4" tabindex="-1"></a>  <span class="at">--input_path</span> <span class="dt">\</span></span>
<span id="cb27-5"><a href="s-analysis.html#cb27-5" tabindex="-1"></a>  <span class="at">--save_path</span> <span class="dt">\</span></span>
<span id="cb27-6"><a href="s-analysis.html#cb27-6" tabindex="-1"></a>  <span class="at">--flowcell</span> FLO-MIN106 <span class="dt">\</span></span>
<span id="cb27-7"><a href="s-analysis.html#cb27-7" tabindex="-1"></a>  <span class="at">--kit</span> SQK-RBK004s</span></code></pre></div>
<p>As we can see from the command line this tool requires the flowcell model and eventually barcoding kit information.</p>
</div>
<div id="dorado" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Dorado<a href="s-analysis.html#dorado" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Recently released by Nanopore, Dorado is a high-performance, easy-to-use, open source basecaller for Oxford Nanopore reads, with the options of super, high and low, accuracy.</p>
<p><a href="https://github.com/nanoporetech/dorado">https://github.com/nanoporetech/dorado</a></p>
<p>First download the flowcell model kit (You need to know which flowcell was used):</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="s-analysis.html#cb28-1" tabindex="-1"></a><span class="ex">dorado</span> download <span class="at">--model</span> dna_r10.4.1_e8.2_400bps_hac@v4.1.0</span></code></pre></div>
<p>Dorado usage:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="s-analysis.html#cb29-1" tabindex="-1"></a><span class="ex">dorado</span> basecaller <span class="dt">\</span></span>
<span id="cb29-2"><a href="s-analysis.html#cb29-2" tabindex="-1"></a>  <span class="at">-b</span> 36 <span class="dt">\</span></span>
<span id="cb29-3"><a href="s-analysis.html#cb29-3" tabindex="-1"></a>  <span class="at">--device</span> cpu <span class="dt">\</span></span>
<span id="cb29-4"><a href="s-analysis.html#cb29-4" tabindex="-1"></a>  <span class="at">--emit-fastq</span></span>
<span id="cb29-5"><a href="s-analysis.html#cb29-5" tabindex="-1"></a>  <span class="ex">mysample.FAST5</span></span></code></pre></div>
</div>
</div>
<div id="filtering-and-trimming" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Filtering and Trimming<a href="s-analysis.html#filtering-and-trimming" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The fastq file containing the 16S sequence need to be filtered based on quality and/or read length, and optional trimmed after passing filter</p>
<div id="nanofilt" class="section level3 hasAnchor" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> NanoFilt<a href="s-analysis.html#nanofilt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>NanoFilt - filtering and trimming of long read sequencing data</p>
<p><a href="https://github.com/wdecoster/nanofilt">https://github.com/wdecoster/nanofilt</a></p>
<p><em>Requirement</em></p>
<p>To execute this tool, you need to activate the conda environment</p>
<p><code>mamba activate /home/irsa/miniconda3/envs/ONTpp</code></p>
<p><strong>STEP TIPS:</strong></p>
<ul>
<li>Your input data are available in the following path: <code>/SERVER/16s_data/</code></li>
<li>Your task should be to use NanoFilt on each fastq file, using the following options:
<ul>
<li><code>--length LENGTH</code> Filter on a minimum read length</li>
<li><code>--maxlength MAXLENGTH</code> Filter on a maximum read length</li>
<li><code>--quality QUALITY</code> Filter on a minimum average read quality score</li>
</ul></li>
<li>Create the folder for your output files (to use in your script)</li>
<li>To perform this step you should create a script (Recommended name: <code>s01_nanofilt.sh</code>), and give it permission to be executed: <code>chmod u+x s01_nanofilt.sh</code></li>
<li>Execute your script as follow: <code>./s01_nanofilt.sh</code></li>
<li>Look at the results</li>
</ul>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s01_nanofilt.sh</code></p>
<p><code>touch s01_nanofilt.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="s-analysis.html#cb30-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb30-2"><a href="s-analysis.html#cb30-2" tabindex="-1"></a></span>
<span id="cb30-3"><a href="s-analysis.html#cb30-3" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> /SERVER/16s_data/<span class="pp">*</span>fastq</span>
<span id="cb30-4"><a href="s-analysis.html#cb30-4" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb30-5"><a href="s-analysis.html#cb30-5" tabindex="-1"></a>    <span class="va">f</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span> .fastq<span class="va">)</span></span>
<span id="cb30-6"><a href="s-analysis.html#cb30-6" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span></span>
<span id="cb30-7"><a href="s-analysis.html#cb30-7" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span></span>
<span id="cb30-8"><a href="s-analysis.html#cb30-8" tabindex="-1"></a>    <span class="fu">cat</span> <span class="va">$i</span> <span class="kw">|</span> <span class="ex">NanoFilt</span> <span class="at">-q</span> 9 <span class="at">-l</span> 1200 <span class="at">--maxlength</span> 1800 <span class="op">&gt;</span> /home/irsa/analisi_16s/output_s01/<span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span>-nf.fastq</span>
<span id="cb30-9"><a href="s-analysis.html#cb30-9" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_16s/output_s01/</code></pre>
<p>Change its permission:
<code>chmod u+x s01_nanofilt.sh</code></p>
<p>Execute it:
<code>./s01_nanofilt.sh</code></p>
</details>
</div>
</div>
<div id="subsetting" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Subsetting<a href="s-analysis.html#subsetting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are going to reduce the number of reads in order to use less resources for our practical analysis. <strong>It is important to note that this step is not part of a common pipeline</strong>.</p>
<div id="bbmap-tools" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> BBMAP Tools<a href="s-analysis.html#bbmap-tools" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>BBMap - short read aligner for DNA/RNAseq, and other bioinformatic tools, including BBMap.</p>
<p><a href="https://github.com/BioInfoTools/BBMap">https://github.com/BioInfoTools/BBMap</a></p>
<p>From this step, we use a script provided by BBMAP tools collection, called: <code>reformat.sh</code>, which reformats reads to change ASCII quality encoding, interleaving, file format, or compression format.</p>
<p><em>Requirement</em></p>
<p>To execute this tool, you need to activate the conda environment:</p>
<p><code>mamba activate /home/irsa/miniconda3/envs/ONTpp</code></p>
<p><strong>STEP TIPS:</strong></p>
<ul>
<li>Your input data should be available from the output folder of the previous step</li>
<li>Your task should be to use <code>reformat.sh</code> on <strong>each output files obtained from the previous step</strong>, using the following options:
<ul>
<li><code>in=&lt;file&gt;</code> Input file</li>
<li><code>out=&lt;outfile&gt;</code> Ouput file</li>
<li><code>samplereadstarget=10000</code> Exact number of OUTPUT reads (or pairs) desired.</li>
</ul></li>
<li>Create the folder for your output files (to use in your script)</li>
<li>To perform this step you should create a script (Recommended name: <code>s02_subsampling.sh</code>), and give it permission to be executed: <code>chmod u+x s02_subsampling.sh</code></li>
<li>Execute your script as follow: <code>./s02_subsampling.sh</code></li>
<li>Look at the results</li>
</ul>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s02_subsampling.sh</code></p>
<p><code>touch s02_subsampling.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="s-analysis.html#cb32-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb32-2"><a href="s-analysis.html#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="s-analysis.html#cb32-3" tabindex="-1"></a><span class="co"># Create output directory for this script</span></span>
<span id="cb32-4"><a href="s-analysis.html#cb32-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> /home/irsa/analisi_16s/output_s02/</span>
<span id="cb32-5"><a href="s-analysis.html#cb32-5" tabindex="-1"></a></span>
<span id="cb32-6"><a href="s-analysis.html#cb32-6" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> /home/irsa/analisi_16s/output_s01/<span class="pp">*</span>-nf.fastq</span>
<span id="cb32-7"><a href="s-analysis.html#cb32-7" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb32-8"><a href="s-analysis.html#cb32-8" tabindex="-1"></a>    <span class="va">f</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span> <span class="at">-nf.fastq</span><span class="va">)</span></span>
<span id="cb32-9"><a href="s-analysis.html#cb32-9" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span></span>
<span id="cb32-10"><a href="s-analysis.html#cb32-10" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span></span>
<span id="cb32-11"><a href="s-analysis.html#cb32-11" tabindex="-1"></a>    <span class="ex">reformat.sh</span> in=<span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span> out=/home/irsa/analisi_16s/output_s02/<span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span>-ss-nf.fastq samplereadstarget=10000</span>
<span id="cb32-12"><a href="s-analysis.html#cb32-12" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_16s/output_s02/</code></pre>
<p>Change its permission:
<code>chmod u+x s02_subsampling.sh</code></p>
</details>
</div>
</div>
<div id="taxonomic-assignment" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Taxonomic Assignment<a href="s-analysis.html#taxonomic-assignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Last step, the sequences are compared to a reference database for taxonomic assignment and a relative abundance estimator for 16S genomic sequences.</p>
<div id="emu" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> EMU<a href="s-analysis.html#emu" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Emu - species-level taxonomic abundance for full-length 16S reads.</p>
<p>This tool use a method optimized for error-prone full-length reads. However, it can be used for short-reads.</p>
<p><a href="https://github.com/treangenlab/emu">https://github.com/treangenlab/emu</a></p>
<p>To perform this annotation, we need a reference database that contains all the taxonomic information, which was already been downloaded in the following path: <code>/SERVER/emu_database</code></p>
<p><em>Requirement</em></p>
<p>To execute this tool, you need to activate the conda environment:</p>
<p><code>mamba activate /home/irsa/miniconda3/envs/emu</code></p>
<p><strong>STEP TIPS:</strong></p>
<ul>
<li>Your input data should be available from the previous step</li>
<li>Your task should be to use <code>emu abundance</code> on <strong>each output files obtained from the previous step</strong>, using the following options:
<ul>
<li><code>--type map-ont</code> denote sequencer [short-read:sr, Pac-Bio:map-pb, ONT:map-ont]</li>
<li><code>--keep-counts</code> include estimated read counts for each species in output</li>
<li><code>--output-dir &lt;output_dir&gt;</code> directory for output results (to be created in advance)</li>
<li><code>--output-basename &lt;basename_files&gt;</code> basename of all output files saved in output-dir; default utilizes basename from input file(s)</li>
</ul></li>
<li>Create the folder for your output files (to use in your script)</li>
<li>To perform this step you should create a script (Recommended name: <code>s03_abundance.sh</code>), and give it permission to be executed: <code>chmod u+x s03_abundance.sh</code>
<ul>
<li>Specify the path of the EMU database with the following line as an action: <code>export EMU_DATABASE_DIR=/SERVER/emu_database</code></li>
<li>Last line of your script, should be an action that performs the command <code>emu combine-outputs</code> to create a single table containing all Emu output relative abundances in a single directory. Note this function will select all the .tsv files in the provided directory that contain ‘rel-abundance’ in the filename. Use the following options in your <code>emu combine-outputs</code> command:
<ul>
<li><code>--counts output</code> estimated counts rather than relative abundance percentage in combined table. Only includes Emu relative abundance outputs that already have ‘estimated counts’</li>
<li><code>tax_id</code> to get results for the most specific taxa level</li>
</ul></li>
</ul></li>
<li>Execute your script as follow: <code>./s03_abundance.sh</code></li>
<li>Look at the results</li>
</ul>
<details>
<summary>
<strong>[SPOILER]</strong> - Scripts that we will use
</summary>
<p>We create an empty file called <code>s03_abundance.sh</code></p>
<p><code>touch s03_abundance.sh</code></p>
<p>We can write our actions in the scripts as follows:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="s-analysis.html#cb34-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb34-2"><a href="s-analysis.html#cb34-2" tabindex="-1"></a></span>
<span id="cb34-3"><a href="s-analysis.html#cb34-3" tabindex="-1"></a><span class="co"># mamba activate /home/irsa/miniconda3/envs/emu</span></span>
<span id="cb34-4"><a href="s-analysis.html#cb34-4" tabindex="-1"></a></span>
<span id="cb34-5"><a href="s-analysis.html#cb34-5" tabindex="-1"></a><span class="bu">export</span> <span class="va">EMU_DATABASE_DIR</span><span class="op">=</span>/SERVER/emu_database</span>
<span id="cb34-6"><a href="s-analysis.html#cb34-6" tabindex="-1"></a></span>
<span id="cb34-7"><a href="s-analysis.html#cb34-7" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> /home/irsa/analisi_16s/output_s02/<span class="pp">*</span>-ss-nf.fastq</span>
<span id="cb34-8"><a href="s-analysis.html#cb34-8" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb34-9"><a href="s-analysis.html#cb34-9" tabindex="-1"></a>    <span class="va">f</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span> <span class="at">-ss-nf.fastq</span><span class="va">)</span></span>
<span id="cb34-10"><a href="s-analysis.html#cb34-10" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span></span>
<span id="cb34-11"><a href="s-analysis.html#cb34-11" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span></span>
<span id="cb34-12"><a href="s-analysis.html#cb34-12" tabindex="-1"></a>    <span class="ex">emu</span> abundance <span class="st">&quot;</span><span class="va">$i</span><span class="st">&quot;</span> <span class="at">--type</span> map-ont <span class="at">--output-basename</span> <span class="st">&quot;</span><span class="va">$f</span><span class="st">&quot;</span> <span class="at">--keep-counts</span> <span class="at">--output-dir</span> /home/irsa/analisi_16s/output_s03/ <span class="at">--threads</span> 4</span>
<span id="cb34-13"><a href="s-analysis.html#cb34-13" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb34-14"><a href="s-analysis.html#cb34-14" tabindex="-1"></a></span>
<span id="cb34-15"><a href="s-analysis.html#cb34-15" tabindex="-1"></a></span>
<span id="cb34-16"><a href="s-analysis.html#cb34-16" tabindex="-1"></a><span class="ex">emu</span> combine-outputs <span class="at">--counts</span> /home/irsa/analisi_16s/output_s03/ tax_id</span>
<span id="cb34-17"><a href="s-analysis.html#cb34-17" tabindex="-1"></a></span></code></pre></div>
<p>Create output directory for this script (Change <code>irsa</code> with your <code>utenteX</code> name)</p>
<pre><code>mkdir -p /home/irsa/analisi_16s/output_s03/</code></pre>
<p>Change its permission:
<code>chmod u+x s03_abundance.sh</code></p>
</details>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-the-command-line.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="metagenomics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["icme13.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
